zap-widget/src/components/SearchBox.tsx

```diff
@@ -0,0 +1,172 @@
+import decode from 'light-bolt11-decoder'
+import { useEffect, useState } from 'react'
+import { useClient } from '../context'
+import SimpleBoostWrapper from './SimpleBoostWrapper'
+
+const boosts = [
+  { id: 'pkg1', amount: 1 },
+  { id: 'pkg2', amount: 5 },
+  { id: 'pkg3', amount: 10 },
+  { id: 'pkg4', amount: 25 },
+  { id: 'pkg5', amount: 50 },
+  { id: 'pkg6', amount: 100 },
+]
+
+type ResultType = {
+  packageName?: string
+  description?: string
+  warn?: boolean
+  hint?: string
+  lnAddress?: string
+}
+
+function SearchBox() {
+  const [packageQueryName, setPackageQueryName] = useState('')
+  const [result, setResult] = useState<ResultType | null>(null)
+  const [isLoading, setIsLoading] = useState(false)
+  const [amountInSats, setAmountInSats] = useState(0)
+
+  const { invoice, setInvoice } = useClient()
+
+  useEffect(() => {
+    const decodeInv = () => {
+      if (invoice.length > 1) {
+        const amountInSat = decode.decode(invoice) as { sections: { value?: string | number }[] }
+        setAmountInSats(+amountInSat.sections[2]?.value / 1000)
+      }
+    }
+
+    decodeInv()
+  }, [invoice])
+
+  const fetchPackage = async () => {
+    try {
+      //clear the result and input alone - when the cncel icon is clicked
+      if (result) {
+        setResult(null)
+        setPackageQueryName('')
+        setInvoice('')
+        setAmountInSats(0)
+        return
+      }
+
+      //fetch the package
+      setIsLoading(true)
+      if (packageQueryName.length > 1) {
+        const response = await fetch(`https://registry.npmjs.org/${packageQueryName}/latest`)
+        const packageInfo = await response.json()
+        console.log(packageInfo)
+
+        if (!packageInfo || packageInfo === 'Not Found') {
+          setResult({
+            warn: true,
+            packageName: packageQueryName,
+            hint: '⚠️ Can’t find this package. Typo or doesn’t exist? 🤔',
+          })
+          return
+        } else if (!packageInfo.funding || packageInfo.funding.type !== 'lightning') {
+          setResult({
+            warn: true,
+            packageName: packageQueryName,
+            hint: '⚠️ This package does not accept bitcoin tips yet 🥺',
+          })
+          return
+        } else {
+          const lnAddress = packageInfo.funding.url
+
+          setResult({
+            warn: false,
+            lnAddress: lnAddress.startsWith('lightning:') ? lnAddress.substring(10) : lnAddress,
+            packageName: packageInfo.name,
+            description: packageInfo.description,
+          })
+        }
+      }
+    } catch (err) {
+      console.error('Error fetching package:', err)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  return (
+    <div className="flex flex-col items-center justify-center min-h-screen mx-8">
+      <div className="w-full flex items-center justify-center flex-col md:flex-row gap-3">
```

Comment:
This should be `gap-2 md:gap-6` (`8px` in mobile and `24px` in bigger screens)

In Figma, you can click on an element and hold `option` key and hover on it's corresponding elements to find the distance between them

<img width="644" height="394" alt="Screenshot 2025-07-22 at 4 36 29 PM" src="https://github.com/user-attachments/assets/8d73145b-5205-4ae1-abd6-c6f84a9e4434" />